algorithms/48/fib.cpp

```diff
@@ -1,11 +1,35 @@
 #include <iostream>
+#include <bits/stdc++.h>
 using namespace std;
 
+void mult(long long A[2][2], long long B[2][2]){
+	long long q,w,e,r;
+	q=A[0][0]*B[0][0]+A[0][1]*B[1][0];
+	w=A[0][0]*B[0][1]+A[0][1]*B[1][1];
+	e=A[1][0]*B[0][0]+A[1][1]*B[1][0];
+	r=A[1][0]*B[0][1]+A[1][1]*B[1][1];
+	A[0][0]=q;
+	A[0][1]=w;
+	A[1][0]=e;
+	A[1][1]=r;
+}
+
+void matrix_pow(long long A[2][2], int n){
+	long long M[2][2]={{1,1},{1,0}};
+	for(int i=2;i<n;i++){
+		mult(A, M);
+	}
```

Comment:
Hhm this is correct, though it could be still optimized. Up to you if you want to further optimize here

Reply:
Optimised to recursively go through matrix_pow